addons/gamification/models/goal.py

```diff
@@ -285,7 +285,7 @@ def update_goal(self):
                     safe_eval(code, cxt, mode="exec", nocopy=True)
                     # the result of the evaluated codeis put in the 'result' local variable, propagated to the context
                     result = cxt.get('result')
-                    if result is not None and isinstance(result, (float, pycompat.integer_types)):
+                    if result is not None and isinstance(result, (float, int)):
```

Comment:
Not linked directly to this PR but I think we can remove the first check: `None` is not an instance of `float` or `int`. It's unlikely this code is performance-critical so it's not like we need the performance gain of skipping `isinstance` on a `None` result.